javers-spring-boot-starter-sql/src/main/java/org/javers/spring/boot/aot/JaversNativeHintsConfiguration.java

```diff
@@ -0,0 +1,170 @@
+package org.javers.spring.boot.aot;
+
+import org.springframework.aot.hint.MemberCategory;
+import org.springframework.aot.hint.RuntimeHints;
+import org.springframework.aot.hint.RuntimeHintsRegistrar;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.ImportRuntimeHints;
+
+import java.util.Arrays;
+
+@Configuration
+@ImportRuntimeHints(JaversNativeHintsConfiguration.ApplicationRuntimeHints.class)
+public class JaversNativeHintsConfiguration {
+    static class ApplicationRuntimeHints implements RuntimeHintsRegistrar {
+        @Override
+        public void registerHints(RuntimeHints hints, ClassLoader classLoader) {
+
+            hints.resources().registerResourceBundle("org.aspectj.weaver.weaver-messages");
+
+            var classes = Arrays.asList(
+                    "java.util.ArrayList",
+
+                    // Package: org.javers.core
+                    "org.javers.core.Javers",
```

Comment:
1. hardcoding all class names seems fragile, what if we change something?

2. Could you please explain (or point to some docs) why do we need to register all these names to achieve Spring Native support 

3. how this move affects classic jvm apps?

Reply:
Hi bartoszwalacik,

1.
hardcoding class names is unfortunately exactly how native hints work ... there is basically no other alternative than this.
You need to tell the compiler upfront which classes need to be registered for reflection, so that they are prepared during compilation.
If classes are changed a Runtime Exception will be thrown as the Class.Forname fails then.
Normally classes are listed as Classes, and not as literals, which would automatically change upon Renaming the class.

Unfortunately Javers uses a lot of Package protected classes, that are not importable this way.

2.
See 1. .. the compiler needs to be told upfront which classes use reflection and all of these classes do so.
https://www.baeldung.com/spring-native-intro

3.
It should not affect jvm apps at all

Reply:
@goafabric thanks for your answer, still not sure I fully understand it, most Javers' classes are , well, just normal classes, only a few of them, or maybe only the one -- ReflectionUtil -- uses reflection API, so why registering them all?

Reply:
concerning 3. , from what I see, this RuntimeHintsRegistrar is executed unconditionally
so it will slow down starting Javers for normal JVM apps. Could you please come up with some condition, end execute it only when needed by a native image app? might is also be an alternative starter

Reply:
The RuntimeRegistrar is only executed during build time, never during run time.
You may validate that by setting a break point.
And then also only if the plugin "org.graalvm.buildtools.native" is inside the build.

Concerning Reflection, it is less about classes that use ReflectionUtil and more about which classes are accessed reflectively.
This will be every ClassForname(class) and every field, method, constructor etc that will be executed via reflection.

The list was initially generated the graalvm agent.
And after that you have to add manually further classes the agent did not catch.

--- example---
class A() {
  doReflectionStuff() {
     class.forName(B.class);
     class.forName(C.class);
     class.forName(D.class);
  }
}

=> needs reflection hints for B,C + D, not for A

Reply:
got it, thanks for the explanation, last thing before merge, could you please add this AOT support also the the javers Mongo starter? (see https://github.com/javers/javers/blob/master/javers-spring-boot-starter-mongo/src/main/java/org/javers/spring/boot/mongo/JaversMongoAutoConfiguration.java)

Reply:
thank you .. i had the same idea, but unfortunately it's not so easy done, 
as i would have to do them process of figuring out the right hints for mongodb again .. don't think this will be 1:1 copy ..
so my idea was to first get the JPA stuff merged and then eventually taking a second look when I find some time

Reply:
thanks, maybe we can move this large array of class names to the javers-spring package and share for two starters (no duplication then). and just a question, how can I refresh this array of class names when sth changes?

Reply:
ok I took the extra time for the refactoring .. I hope we can now getting it merged ...

Reply:
thanks @goafabric , well done, I will merge and release soon